### ISSUE TITLE
`Image`'s width and height props throw warning of invalid type on '100%' value

### ISSUE THREAD

**Issue by idrakimuhamad** [tamagui/tamagui#900]
The `Image` and `Avatar.Image` component will throw failed prop type if the width and height were given a "100%" value, despite the prop type accepting number and string.

```jsx
<Image src={src} width="100%" height="100%" />
```

```bash
 Warning: Failed prop type: Invalid prop `height` of type `string` supplied to `Image`, expected `number`. at BaseImage
```

I think in React Native's own Image component, "100%" is an acceptable value for both `width` and `height`. Also, rendering wise, the image are rendered correctly. But the warning are distraction and also could break our E2E test when the yellow box appearing.

**Comment by natew** [comment]
What platform are you seeing this on?

**Comment by natew** [comment]
<img width="459" alt="CleanShot 2023-04-05 at 18 23 53@2x" src="https://user-images.githubusercontent.com/12100/230271636-596c572d-b6fc-49e7-8a72-1dad295d342f.png">

<img width="354" alt="CleanShot 2023-04-05 at 18 24 20@2x" src="https://user-images.githubusercontent.com/12100/230271678-d3050ba2-0693-4ef9-b1fd-4ee8eea018c6.png">


React Native Web and React both seem to document it being a number, can you repro?

**Comment by idrakimuhamad** [comment]
iOS and Android on the Logbox. Didn't check on Web.

The type for `@tamagui/Image` `BaseProps` seems indicating otherwise.

![Screenshot 2023-04-06 at 12 25 37 PM](https://user-images.githubusercontent.com/1400323/230271805-b32f9710-f5a1-4c40-a161-facbac955600.png)

**Comment by natew** [comment]
Fix going out in a minute 8d5c85614d84a7348da4ea4d2709b0c6fb4a7ca1
